README.md

```diff
@@ -355,6 +356,75 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+#### Why use Interface instead of Type for typing props?
+
+`interface` are a little more powerful than the corresponding `type` declaration.
+
+One of the reasons to use `interface` is when you need to compose two or more types, you have the
+option of extending those types with an interface, or intersecting them in a type alias. TS team
+says:
+
+> Interfaces create a single flat object type that detects property conflicts, which are usually
+> important to resolve! Intersections on the other hand just recursively merge properties, and in
+> some cases produce never. Interfaces also display consistently better, whereas type aliases to
+> intersections can't be displayed in part of other intersections. Type relationships between
+> interfaces are also cached, as opposed to intersection types as a whole. A final noteworthy
+> difference is that when checking against a target intersection type, every constituent is checked
+> before checking against the "effective"/"flattened" type. For this reason, extending types with
+> interfaces/extends is suggested over creating intersection types.
+
+This is preferred:
+
+```ts
+interface Foo extends Bar, Baz {
+  someProp: string;
+}
+```
+
+Instead of this:
+
+```ts
+type Foo = Bar &
+  Baz & {
+    someProp: string;
+  };
+```
+
+#### Why not use React.FC or FunctionComponent type?
+
+- Provides an implicit definition of `children`
+- Doesn't support generics.
+- Doesn't work correctly with `defaultProps`
+
+#### Implicit or explicit `returnType`?
+
+It depends. If we know we want to return a JSX element, it is not necessary to type the `returnType`
+because it is inferred from what you actually return, so can be omitted most of the time.
```

Comment:
For things in the public API it is better to always set the return type. In case you make a mistake (in a refactoring),  you immediately see what is wrong

Reply:
@ThijsTyZ 

Yeah. For that explanation I state this extra text:

_But if you may going to return something other than JSX, then define the `returnType`._

Reply:
I don't think there should be an exception for components that return a `React.ReactElement`

Reply:
Agree, both for consistency and speed I would prefer to always specify the component return type.
Also makes it easier if you accidentally return something invalid.